marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mastering.sjs

```diff
@@ -0,0 +1,102 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const DataHubSingleton = require('/data-hub/5/datahub-singleton.sjs');
+
+// define constants for caching expensive operations
+const dataHub = DataHubSingleton.instance();
+
+const collections = ['http://marklogic.com/data-hub/steps/mastering', 'http://marklogic.com/data-hub/steps'];
+const databases = [dataHub.config.STAGINGDATABASE, dataHub.config.FINALDATABASE];
+const permissions = [xdmp.permission(dataHub.consts.DATA_HUB_FLOW_WRITE_ROLE, 'update'), xdmp.permission(dataHub.consts.DATA_HUB_FLOW_READ_ROLE, 'read')];
+const requiredProperties = ['name'];
+
+function getNameProperty() {
+    return 'name';
+}
+
+function getEntityNameProperty() {
```

Comment:
Mastering/matching/merging actually use "targetEntity" instead of targetEntityType - that's something we'll be making consistent in 5.4. But while this should be "targetEntity", I think you can safely remove this function altogether, as it's not exported below. Can also remove - getSelectedSourceProperty and getVersionProperty (getVersionProperty really needs to be removed in a separate JIRA ticket altogether, as "version" isn't being used by artifacts yet).

Reply:
Removing getVersionProperty() causes this error:

RESTAPI-INVALIDCONTENT: (err:FOER0000) Invalid content: invalid mlRunIngest extension: could not parse JavaScript extension mlRunIngest; please see the server error log for detail JS-JAVASCRIPT: getVersionProperty, -- Error running JavaScript request: ReferenceError: getVersionProperty is not defined; mlRunIngest either is not a valid JavaScript module or does not export functions (transform)

Reply:
That error happens during bootstrap